content/algorithms/Dijkstra/Dijkstra.md

```diff
@@ -0,0 +1,241 @@
+from heapq import heappop as pop
+from itertools import count
+
+'''
+So, the graph 'flight_path' is already defined with all nodes and edges. Now, before implementing the algorithm, 
+one first needs to make the data available in the proper format to access.
+
+The first thing to do is to have a dictionary with every node as keys and (connecting node, weight) as values so that one can
+traverse easily.
+'''
+flight_succ = flight_path._succ if flight_path.is_directed() else flight_path._adj
+
+# we need to extract the distance between 2 nodes from the graph and for that we need to define weight function
+def _weight_function(G, weight):
+    return lambda u, v, data: data.get(weight, 1)
+
+weight = _weight_function(flight_path, "weight")
+
+'''
+The next step is to define various dictionaries to store and track all nodes path.
+'''
+dist = {}  # dictionary of final distances
+seen = {}  # dictionary of visited nodes with recent shortest distance
+paths= {}  # dictionary to store path list
+
+# fringe is heapq with 3-tuples (distance,c,node)
+# use the count c to avoid comparing nodes (may not be able to)
+c = count()
+fringe = []
+
+# we want to find the shortest distance from DEL to LCY
+source='DEL'
+target='LCY'
+paths[source]=[source] 
+
+# Now, as I said earlier we'll assign the source node 0 value
+seen[source] = 0 
+push(fringe, (0, next(c), source))
+
+'''
+It's time to start traversing the graph starting from source node.
+'''
+
+while fringe: 
+        (d, _, v) = pop(fringe) # d will store the distance of the node and v will store the node name
+        if v in dist:
+            continue  # already searched this node.      
+        dist[v] = d 
+        if v == target: 
+            break
+           
+        # traversing the neighbours of the node
+        for u, e in flight_succ[v].items(): 
+            distance = weight(v, u, e)
+            
+            if distance is None:
+                continue
+            # vu_dist stores the total distance from source node to u. Like, if v=ZRH and u= GVA, then vu_dist = 8    
+            vu_dist = dist[v] + distance 
+            
+            '''
+            If u is already in dist then there can be 2 cases, either the graph has negative cycle or there might
+            be another shortest path to u.
+            '''
+            if u in dist:
+                u_dist = dist[u]
+                if vu_dist < u_dist:
+                    raise ValueError("Contradictory paths found:", "negative weights?")
+                    
+            # updating the new shortest distance and adding the next node to visit
+            elif u not in seen or vu_dist < seen[u]:
+                seen[u] = vu_dist
+                push(fringe, (vu_dist, next(c), u))
+                if paths is not None:
+                    paths[u] = paths[v] + [u]
+
+# printing the distance and path from the source node 'DEL' to target node 'LCY'                    
+print(dist[target], paths[target])
+```
```

Comment:
BTW we don't need to implement the algorithm in the same exact way as done inside NetworkX for nx-guides. The goal is to provide a pedagogical source so if possible feel free to remove bits that you think can be better compressed :)

Reply:
So, can I replace it with a basic general implementation?